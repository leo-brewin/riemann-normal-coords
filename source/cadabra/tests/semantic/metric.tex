\documentclass[12pt]{cdblatex}

\begin{document}

\section*{\jobname}

\CdbSetup{action=hide}

\begin{cadabra}
   import shared

   import cdblib

   term00A = cdblib.get ('check000','expected/metric.json')
   term01A = cdblib.get ('check001','expected/metric.json')
   term02A = cdblib.get ('check002','expected/metric.json')
   term03A = cdblib.get ('check003','expected/metric.json')
   term04A = cdblib.get ('check004','expected/metric.json')
   term05A = cdblib.get ('check005','expected/metric.json')
   term06A = cdblib.get ('check005','expected/metric.json')
   term07A = cdblib.get ('check005','expected/metric.json')

   term00B = cdblib.get ('check000','output/metric.json')
   term01B = cdblib.get ('check001','output/metric.json')
   term02B = cdblib.get ('check002','output/metric.json')
   term03B = cdblib.get ('check003','output/metric.json')
   term04B = cdblib.get ('check004','output/metric.json')
   term05B = cdblib.get ('check005','output/metric.json')
   term06B = cdblib.get ('check005','output/metric.json')
   term07B = cdblib.get ('check005','output/metric.json')

   # bug: can't push this function into shared.py
   #      no synatx error, but cadabra doesn't cancel equal terms
   # see ~/cadabra/bugs/bug02

   def check (objA,objB):
       tmp := @(objA) - @(objB).
       distribute         (tmp)
       tmp = shared.standard_indices (tmp)
       tmp = shared.product_sort (tmp)
       rename_dummies     (tmp)
       canonicalise       (tmp)

       return tmp

   diff000 = shared.check (term00A,term00B)   # cdb (diff000,diff000)
   diff001 = shared.check (term01A,term01B)   # cdb (diff001,diff001)
   diff002 = shared.check (term02A,term02B)   # cdb (diff002,diff002)
   diff003 = shared.check (term03A,term03B)   # cdb (diff003,diff003)
   diff004 = shared.check (term04A,term04B)   # cdb (diff004,diff004)
   diff005 = shared.check (term05A,term05B)   # cdb (diff005,diff005)
   diff006 = shared.check (term06A,term06B)   # cdb (diff006,diff006)
   diff007 = shared.check (term07A,term07B)   # cdb (diff007,diff007)

\end{cadabra}

\begin{dgroup*}
   \Dmath*{ \cdb*{diff000} }
   \Dmath*{ \cdb*{diff001} }
   \Dmath*{ \cdb*{diff002} }
   \Dmath*{ \cdb*{diff003} }
   \Dmath*{ \cdb*{diff004} }
   \Dmath*{ \cdb*{diff005} }
   \Dmath*{ \cdb*{diff006} }
   \Dmath*{ \cdb*{diff007} }
\end{dgroup*}

\end{document}
