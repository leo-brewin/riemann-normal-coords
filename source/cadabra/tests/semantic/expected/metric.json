{
  "check000": "g_{a b} -  1/3 x^{c} x^{d} R_{a c b d} -  1/6 x^{c} x^{d} x^{e} \\partial_{c}(R_{a d b e})",
  "check001": "g_{a b} -  1/3 x^{c} x^{d} R_{a c b d} -  1/6 x^{c} x^{d} x^{e} \\partial_{c}(R_{a d b e}) +  1/360 x^{c} x^{d} x^{e} x^{f} (-3R_{b c d g} R^{g}_{f a e}-13R_{a c d g} R^{g}_{f b e}-9g_{b g} \\partial_{c d}(R^{g}_{f a e})-9g_{a g} \\partial_{c d}(R^{g}_{f b e})) +  1/180 x^{c} x^{d} x^{e} x^{f} x^{g} (-3R^{h}_{d a c} \\partial_{e}(R_{b f g h})-R_{b c d h} \\partial_{e}(R^{h}_{g a f})-3R^{h}_{d b c} \\partial_{e}(R_{a f g h})-g_{b h} \\partial_{c d e}(R^{h}_{g a f})-R_{a c d h} \\partial_{e}(R^{h}_{g b f})-g_{a h} \\partial_{c d e}(R^{h}_{g b f}))",
  "check002": "g_{a b} -  1/3 x^{c} x^{d} R_{a c b d} -  1/6 x^{c} x^{d} x^{e} \\nabla_{c}(R_{a d b e}) +  1/180 x^{c} x^{d} x^{e} x^{f} (8g^{g h} R_{a c d g} R_{b e f h}-9\\nabla_{c d}(R_{a e b f})) +  1/90 x^{c} x^{d} x^{e} x^{f} x^{g} (2g^{h i} R_{a c d h} \\nabla_{e}(R_{b f g i}) + 2g^{h i} R_{b c d h} \\nabla_{e}(R_{a f g i})-\\nabla_{c d e}(R_{a f b g}))",
  "check003": "g_{a b}",
  "check004": "-x^{c} x^{d} R_{a c b d}",
  "check005": "-x^{c} x^{d} x^{e} \\nabla_{c}(R_{a d b e})",
  "check006": "x^{c} x^{d} x^{e} x^{f} (8g^{g h} R_{a c d g} R_{b e f h}-9\\nabla_{c d}(R_{a e b f}))",
  "check007": "x^{c} x^{d} x^{e} x^{f} x^{g} (2g^{h i} R_{a c d h} \\nabla_{e}(R_{b f g i}) + 2g^{h i} R_{b c d h} \\nabla_{e}(R_{a f g i})-\\nabla_{c d e}(R_{a f b g}))"
}
